Makefile

```diff
@@ -25,27 +25,48 @@ help: ## Display this help message.
 build.%: ## Builds the given docker service. Example: make build.datastore
 	docker compose build $*
 
-build: ## Builds the all docker services.
+build: ## Builds the all docker setup.
+	$(MAKE) install_node
 	docker compose build
 
 run-devserver: # Runs the dev-server
-	docker compose up dev-server -d --no-deps
-	$(MAKE) update.requirements
 	docker compose up angular-build -d
 	$(MAKE) update.package
+	docker cp oppia-angular-build:/app/oppia/node_modules .
+	docker compose stop angular-build
+	docker compose up dev-server -d --no-deps
+	$(MAKE) update.requirements
 	$(MAKE) run-offline
 
 run-offline: # Runs the dev-server in offline mode
 	docker compose up dev-server -d
 	@printf 'Please wait while the development server starts...\n\n'
 	@while [[ $$(curl -s -o /tmp/status_code.txt -w '%{http_code}' http://localhost:8181) != "200" ]] || [[ $$(curl -s -o /tmp/status_code.txt -w '%{http_code}' http://localhost:8181/community-library) != "200" ]]; do \
-		sleep 5; \
+		printf "▓"; \
+		if [[ "$(prod_env)" = 'true' ]] && [[ -n $$(docker ps -q -f status=exited -f name=webpack-compiler) ]]; then \
+			${SHELL_PREFIX} dev-server python -m scripts.generate_build_directory; \
+		fi; \
+		sleep 1; \
 	done
+	@printf '\n\n'
 	@echo 'Development server started at port 8181.'
 	@echo 'Please visit http://localhost:8181 to access the development server.'
 	@echo 'Check dev-server logs using "make logs.dev-server"'
 	@echo 'Stop the development server using "make stop"'
 
+start-devserver-for-tests: ## Starts the development server for the tests
+	docker compose up dev-server -d
+	@printf 'Please wait while the development server starts...\n\n'
+	@while [[ $$(curl -s -o /tmp/status_code.txt -w '%{http_code}' http://localhost:8181) != "200" ]] || [[ $$(curl -s -o /tmp/status_code.txt -w '%{http_code}' http://localhost:8181/community-library) != "200" ]]; do \
+		printf "▓"; \
+		if [[ "$(prod_env)" = 'true' ]] && [[ -n $$(docker ps -q -f status=exited -f name=webpack-compiler) ]]; then \
+			${SHELL_PREFIX} dev-server python -m scripts.generate_build_directory; \
+		fi; \
+		sleep 1; \
+	done
+	@printf '\n\n'
+	@echo 'Development server started at port 8181.'
```

Comment:
This seems to be similar to `run-offline`. Can we combine them?

Reply:
ah, I added a new make command for starting the server for the tests because run-offline command also prints some logs which are not required in the tests. thats why I added a new make command for the tests